### PR TITLE
Use ExpeditedDelay in with our features that pause while non-visible

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by JoeRobich** [comment]
Retargeting to main so we can remove main-vs-deps branch

**Review by davidwengier — APPROVED** [review]

File: src/EditorFeatures/Core/Workspaces/ITextBufferVisibilityTracker.cs [hunk 1]
```diff
@@ -65,7 +66,7 @@ internal static class ITextBufferVisibilityTrackerExtensions
             try
             {
                 // Listen to when the active document changed so that we startup work on a document once it becomes visible.
-                var delayTask = Task.Delay(timeSpan, cancellationToken);
+                var delayTask = asyncListener.Delay(DelayTimeSpan.NonFocus, cancellationToken);
                 await Task.WhenAny(delayTask, visibilityChangedTaskSource.Task).ConfigureAwait(false);
```
Comment (sharwell) [inline src/EditorFeatures/Core/Workspaces/ITextBufferVisibilityTracker.cs:70]:
```suggestion
                await Task.WhenAny(delayTask, visibilityChangedTaskSource.Task).ConfigureAwait(true);
```

You also don't need the switch to main thread inside the `finally` block.

Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/Workspaces/ITextBufferVisibilityTracker.cs:70]: good point!

**Comment by CyrusNajmabadi** [comment]
Closing out for now.  will try to return to this in the future.

**Pull request closed without merging**